src/app/components/articles-grid.tsx

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { usePostQuery } from '../store/hooks/usePostQuery';
+import Article from './article';
+
+const ArticlesGrid = () => {
+    const postQuery = usePostQuery({ page: 0, per_page: 10 });
+
+    if ( postQuery.status === 'pending' ) {
+        return <em>Loading</em>;
+    }
+
+    if ( postQuery.status === 'failed' ) {
```

Comment:
Use `elseif` for better readability.

Reply:
```node
if ( postQuery.status === 'pending' ) {
    return <Loading />;
} else if ( postQuery.status === 'failed' ) {
    return <Message text='e.g. Try again later' />;
} else if ( postQuery.status === 'finished' && ! postQuery.result.posts.length ) {
    return <Message text='e.g. No articles yet' />;
} else {
    return (
        <div>
            {postQuery.result.posts.map((post: Post, i: number) => (
                <Article key={i} id={post.id} />
            ))}
        </div>
    );
}
    
// Or

if ( postQuery.status === 'pending' ) {
    return <Loading />;
}

if ( postQuery.status === 'failed' ) {
    return <Message text='e.g. Try again later' />;
}

if ( postQuery.status === 'finished' && ! postQuery.result.posts.length ) {
    return <Message text='e.g. No articles yet' />;
}

return (
    <div>
        {postQuery.result.posts.map((post: Post, i: number) => (
            <Article key={i} id={post.id} />
        ))}
    </div>
);

Reply:
The second actually looks better given all the scenarios, but we're going to use a loadable/errorbounday component anyways, so you'll only need to care about two conditions.